csrc/scheduler/matmul.cpp

```diff
@@ -525,6 +656,10 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   // Mma object is valid only because cacheBefore has been done on
   //  TV which is not output of MmaOp, as there is an epilogue
   auto mma_result = has_epilogue ? mma->out()->as<TensorView>() : cc;
+  // epilogue shared memory tensor if use shared memory epilogue
+  // mma_result -> cc (if has_epilogue) -> c_smem -> c
+  auto c_smem = params.has_smem_epilogue ? c->cacheBefore() : c;
```

Comment:
I think generally `c_smem` should be `cc->cacheAfter()` instead of `c->cacheBefore()`? According to our discussion in the matmul meeting, we generally unswizzle the mma output immediately before doing other operations. But on the other hand, for HSH kernel, what should we do? If we do the s->h cast before unswizzle, we can save shared memory. cc @drzejan2 Do you have any recommendation?

Reply:
I changed to `cc->cacheAfter()` this allows coalesced loading of other input tensors used in epilogue and simplifies the transform propagation (we don't need to do dryrun swizzle on tensors between mma_result and the output tensor).

Reply:
Let's keep the `cc-cacheAfter()` approach, we shouldn't run out of smem when smem reuse is available and enabled.

I won't be participating in today's matmul meeting, so for me we are good.
I don't want to block this PR which is already open for a long time.